inc/classes/optimization/CSS/class-combine-google-fonts.php

```diff
@@ -113,29 +113,39 @@ protected function find( $pattern, $html ) {
 	 * @return void
 	 */
 	protected function parse( $matches ) {
+		$fonts_array   = [];
+		$subsets_array = [];
 		foreach ( $matches as $match ) {
 			$url   = html_entity_decode( $match[2] );
-			$query = \rocket_extract_url_component( $url, PHP_URL_QUERY );
+			$query = rocket_extract_url_component( $url, PHP_URL_QUERY );
 
 			if ( ! isset( $query ) ) {
 				return;
 			}
 
 			$font = wp_parse_args( $query );
-
-			// Add font to the collection.
-			$this->fonts[] = rawurlencode( htmlentities( $font['family'] ) );
+			if ( isset( $font['family'] ) ) {
+				$font_family = $font['family'];
+				$font_family = rtrim( $font_family, '%7C' );
+				$font_family = rtrim( $font_family, '|' );
+				// Add font to the collection.
+				$fonts_array[] = rawurlencode( htmlentities( $font_family ) );
+			}
 
 			// Add subset to collection.
 			if ( isset( $font['subset'] ) ) {
-				$this->subsets[] = rawurlencode( htmlentities( $font['subset'] ) );
+				$subsets_array[] = rawurlencode( htmlentities( $font['subset'] ) );
 			}
 		}
 
 		// Concatenate fonts tag.
-		$this->subsets = ! empty( $this->subsets ) ? '&subset=' . implode( ',', array_filter( array_unique( $this->subsets ) ) ) : '';
-		$this->fonts   = implode( '|', array_filter( array_unique( $this->fonts ) ) );
-		$this->fonts   = str_replace( '|', '%7C', $this->fonts );
+		$this->subsets = ! empty( $subsets_array ) ? '&subset=' . implode( ',', array_filter( array_unique( $subsets_array ) ) ) : '';
+		if ( ! empty( $fonts_array ) ) {
+			$this->fonts = implode( '|', array_filter( array_unique( $fonts_array ) ) );
+			$this->fonts = str_replace( '|', '%7C', $this->fonts );
```

Comment:
@crystinutzaa Couldn't we implode using `'%7C'` instead of `'|'` to combine these 2 lines of code into one?

Reply:
%7C = rawurlencode( | ), so I think it can be combined into a single implode in there without any str_replace.
I will give it a try and commit if it seems fine

Reply:
Cool. If yes, go ahead and use a ternary to be consistent with subsets line above it.